homeassistant/components/media_player/onkyo.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for Onkyo Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.onkyo/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_SELECT_SOURCE, MediaPlayerDevice)
+from homeassistant.const import STATE_OFF, STATE_ON
+
+REQUIREMENTS = ['https://github.com/danieljkemp/onkyo-eiscp/archive/'
+                'python3.zip#'  'onkyo-eiscp==0.9.2']
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_ONKYO = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Onkyo platform."""
+    from eiscp import eISCP
+    add_devices(OnkyoDevice('{}_{}'.format(
+        receiver.info['model_name'], receiver.info['identifier']), receiver)
```

Comment:
If the name is based on the receiver, why not have this be generated inside the constructor.